### PR TITLE
Fix archive invalidation timezone issue in tracker

### LINKED ISSUE #14298: Tracking can cause requests to option table
Noticed this query is execute some times within a day as part of a tracking request: `SELECT option_name, option_value FROM `option` WHERE option_name LIKE ?`.

image
Output:

```
'2019-03-31'
'UTC'
1554066533'
2019-04-01'
1554076800  
1554076800
```

Stacktrace:
> #0 Piwik\Option->getNameLike(report_to_invalidate_1_2019-03-31%) called at [/var/www/html/core/Option.php:58] #1 Piwik\Option::getLike(report_to_invalidate_1_2019-03-31%) called at [/var/www/html/core/Archive/ArchiveInvalidator.php:70] #2 Piwik\Archive\ArchiveInvalidator->rememberToInvalidateArchivedReportsLater(1, Piwik\Date Object ([*timestamp] => 1554066319,[*timezone] => UTC)) called at [/var/www/html/core/Tracker/Visit.php:561] #3 Piwik\Tracker\Visit->markArchivedReportsAsInvalidIfArchiveAlreadyFinished() called at [/var/www/html/core/Tracker/Visit.php:193] #4 Piwik\Tracker\Visit->handle() called at [/var/www/html/core/Tracker.php:138] #5 

Seems isToday is not quite correctly detected here.

timezone of the website is New York: 
image

### PR DESCRIPTION
**Pull request by katebutler**
Fixes #14298

### CODE REVIEW

File: tests/PHPUnit/Integration/TrackerTest.php [hunk 1]
```diff
@@ -357,6 +359,30 @@ public function test_main_shouldPostEndEvent_EvenIfThereIsAnException()
         $this->assertTrue($called);
     }
 
+    public function test_archiveInvalidation_differentServerAndWebsiteTimezones()
+    {
+        // Server timezone is UTC
+        ini_set('date.timezone', 'America/New_York');
```
Comment (diosmosis) [inline tests/PHPUnit/Integration/TrackerTest.php]: Should this be set down in a tearDown method to make sure the timezone isn't different for other tests?

Comment (tsteur) [reply tests/PHPUnit/Integration/TrackerTest.php]: 👍  in `setUp` we would maybe also need to remember the initial timezone and then restore the original timezone in `tearDown`?

**Comment by diosmosis** [comment]
Code looks good, but an integration test is failing. After it's fixed, this can be merged.

**Comment by katebutler** [comment]
The integration test that was failing when UTC and UTC+5 are different dates has been fixed

File: tests/PHPUnit/Integration/Tracker/VisitTest.php [hunk 2]
```diff
@@ -414,11 +414,12 @@ public function test_markArchivedReportsAsInvalidIfArchiveAlreadyFinished_Should
...
     {
-        $timezone1 = 'UTC+4';
-        $timezone2 = 'UTC+6';
+        // The double-handling below is needed to work around weird behaviour when UTC and UTC+5 are different dates
+        // Example: 4:32am on 1 April in UTC+5 is 11:32pm on 31 March in UTC
+        $midnight = Date::factoryInTimezone('today', 'UTC+5')->setTimezone('UTC+5');
 
-        $currentActionTime1 = Date::today()->setTimezone($timezone1)->getDatetime();
-        $currentActionTime2 = Date::today()->setTimezone($timezone2)->getDatetime();
+        $oneHourAfterMidnight = $midnight->addHour(1)->getDatetime();
```
Comment (tsteur) [inline tests/PHPUnit/Integration/Tracker/VisitTest.php]: those variables are a lot more easier to read and make a lot more sense 👍 This way I would have probably noticed while writing the tests the previous assertions were wrong :)

**Pull request merged**